waku/persistence/sqlite/nwaku.go

```diff
@@ -0,0 +1,82 @@
+package sqlite
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+)
+
+const minSupportedNWAKUversion = 8
+const maxSupportedNWAKUversion = 8
+
+func handleNWakuPreMigration(db *sql.DB) (bool, error) {
+	// Check if there's an user version in the DB, and if migration table does not exist.
+	// Rename existing table, and move data afterwards
+
+	var nwakuDBVersion int
+	err := db.QueryRow("PRAGMA user_version").Scan(&nwakuDBVersion)
+	if err != nil {
+		return false, fmt.Errorf("could not obtain sqlite user_version while attempting to migrate nwaku database: %w", err)
+	}
+
+	var gowakuDBVersion int
+	err = db.QueryRow(`SELECT COUNT(*) FROM schema_migrations`).Scan(&gowakuDBVersion)
+	if err != nil && errors.Is(err, sql.ErrNoRows) {
+		return false, fmt.Errorf("could not obtain schema_migrations data while attempting to migrate nwaku database: %w", err)
+	}
+
+	if nwakuDBVersion == 0 {
+		// not a nwaku db
+		return false, nil
+	}
+
+	if nwakuDBVersion < minSupportedNWAKUversion || nwakuDBVersion > maxSupportedNWAKUversion {
```

Comment:
Seeing there are a good number of conditions here, are these going to go away once nwaku also aligns with same db strategy as go-waku? 
If not, we may have to document somewhere that on noticing abc errors, please delete the db and continue.